src/mqt/bench/benchmark_generation.py

```diff
@@ -303,12 +305,12 @@ def get_native_gates_level(
         # Transpile once more to remove unnecessary gates and optimize the circuit
         compiled = transpile(
             new_qc,
-            basis_gates=gateset.gates,
+            target=gateset,
             optimization_level=opt_level,
             seed_transpiler=10,
         )
     else:
-        compiled = transpile(qc, basis_gates=gateset.gates, optimization_level=opt_level, seed_transpiler=10)
+        compiled = transpile(qc, target=gateset, optimization_level=opt_level, seed_transpiler=10)
```

Comment:
I am fairly sure that we only want to pass the `target.operation_names` to the `basis_gates` kwarg here for the native gates level.
If a regular target were to be passed in, we would also consider its connectivity here.
Please also check this for every other place where the native gates level is used.